paddle/fluid/framework/parallel_executor.cc

```diff
@@ -145,9 +156,9 @@ void ParallelExecutor::BCastParamsToGPUs(
     auto &dims = main_tensor.dims();
     if (paddle::platform::is_gpu_place(main_tensor.place())) {
 #ifdef PADDLE_WITH_CUDA
```

Comment:
I don't think the modify of line159~167 is necessary.
Because if the memory of one Place is insufficient, the program will throw an exception in this [line](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/platform/gpu_info.cc#L119).

Reply:
Actually, the thrown exception will not be handled properly, this PR was submitted to fix this [bug](https://github.com/PaddlePaddle/Paddle/issues/11375)